### DISCUSSION TITLE
Migrate kitty.conf to a standard file format such as TOML or YAML

### DISCUSSION THREAD

**Discussion by exastone** [kovidgoyal/kitty#5904] [Ideas]
Firstly; Is kitty.conf based on a file format? I did some looking and didn't see anything to suggest if it was/ what it is

Secondly; If it's not based on an a common file format, what is the reason? Wouldn't that mean having a custom parser for kitty related .conf files?

Thirdly; If work was done to enable TOML or YAML file format to be used for .conf files, what is the chances of it actually be accepted and merged into master?

**Comment by kovidgoyal** [comment]
On Thu, Jan 19, 2023 at 10:27:05PM -0800, Andrew Stone wrote:
> Firstly; Is kitty.conf based on a file format? I did some looking and didn't see anything to suggest if it was/ what it is

kitty.conf is a file format, the kitty.conf format.

> 
> Secondly; If it's not based on an a common file format, what is the reason? Wouldn't that mean having a custom parser for kitty related .conf files?

So? Why would anyone want some kind of generic file format that cannot
be optimised for a specific use case unless one is too lazy to write a
parser. 

> 
> Thirdly; If work was done to enable TOML or YAML file format to be used for .conf files, what is the chances of it actually be accepted and merged into master?

None, whatsoever. There are zero benefits to using TOML or YAML or
whatever. If you disagree, maybe start by pointing out some beyond the
usual platitudes.

**Reply by exastone** [reply]
who hurt you

**Reply by kovidgoyal** [reply]
You just did, by wasting my time.

**Comment by SolitudeSF** [comment]
you cant unironically suggest that parser for a format slightly more complicated than `ini` is worse than bringing yaml.
